mis_builder/static/src/components/mis_report_widget.esm.js

```diff
@@ -170,6 +187,69 @@ export class MisReportWidget extends Component {
         this.action.doAction(action);
     }
 
+    async _remove_annotation(cell_id) {
+        await this.orm.call(
+            "mis.report.instance.annotation",
+            "remove_annotation",
+            [cell_id, this._instanceId()],
+            {context: this.context}
+        );
+        this.refresh_annotation();
+    }
+
+    async _save_annotation(cell_id) {
+        const text = document.querySelector(".o_mis_builder_annotation_text").value;
+        await this.orm.call(
+            "mis.report.instance.annotation",
+            "set_annotation",
+            [cell_id, this._instanceId(), text],
+            {context: this.context}
+        );
+        await this.refresh_annotation();
+    }
+
+    async annotate(event) {
+        const cell_id = $(event.target).data("cell-id");
+        const note = this.state.mis_report_data.notes[cell_id];
+        const note_text = (note && note.text) || "";
+        var buttons = [
+            {
+                text: this.env._t("Save"),
+                classes: "btn-primary",
+                close: true,
+                click: this._save_annotation.bind(this, cell_id),
+            },
+            {
+                text: this.env._t("Cancel"),
+                close: true,
+            },
+        ];
+        if (typeof note !== "undefined") {
+            buttons.push({
+                text: this.env._t("Remove"),
+                classes: "btn-secondary",
+                close: true,
+                click: this._remove_annotation.bind(this, cell_id),
+            });
+        }
+
+        new Dialog(this, {
+            title: "Annotate",
+            size: "medium",
+            $content: $(
+                qweb.render("mis_builder.annotation_dialog", {
+                    text: note_text,
+                })
+            ),
+            buttons: buttons,
+        }).open();
+    }
+
+    async remove_annotation(event) {
+        const cell_id = $(event.target).data("cell-id");
```

Comment:
Do we really need these $ ? In v18 there is no jquery anymore, so let's not add more dependency on it, that will make migration harder.